### ISSUE TITLE
dask_gateway test fails on catalyst cooperative hub

### ISSUE THREAD

**Issue by sgibson91** [2i2c-org/infrastructure#797]
### Description

The dask gateway tests I implemented in #776 and #796 fail on the catalyst cooperative hub. This is because they self-serve their environment image and it doesn't contain dask gateway.

I thought I could rework the [test in the notebook](https://github.com/2i2c-org/infrastructure/blob/master/deployer/tests/test-notebooks/daskhub/dask_test_notebook.ipynb) to look something like the following:

```python
def test_dask_gateway():
    try:
        import dask_gateway
    except ModuleNotFoundError:
        return None

    # ... Rest of dask gateway testing code here ...

    return None

test_dask_gateway()
```

But I wasn't sure if that would mean we miss cases when dask gateway **should** be there. I guess another option is for the notebook to somehow "know" what hub it is being run on and have a list of hubs that it should allow to pass if dask gateway is not present.

### Value / benefit

We need flexible deployment tests for folk who are serving their own images.

**Comment by sgibson91** [comment]
> it actually makes sense to have this failure

Yep. If we move forward with this strategy though, I'd want to put some work in to parallelise our deployment CI over **hubs**, not just clusters. As it's unfair to communities listed after them in the config file that they don't get updates due to another hubs failure (particularly in the case of self-served images).

> is to migrate them to a basehub

Possibly, I guess we'd need to talk to the community to see how they are using dask?

**Comment by damianavila** [comment]
> Yep. If we move forward with this strategy though, I'd want to put some work in to parallelise our deployment CI over hubs, not just clusters. As it's unfair to communities listed after them in the config file that they don't get updates due to another hubs failure (particularly in the case of self-served images).

💯 , I think this is also super important at the time to speed the CI-powered delivery of fixes on specific hubs (ie. in the middle of one event, we needed to manually run the deployer to save time).

> Possibly, I guess we'd need to talk to the community to see how they are using dask?

Indeed.

**Comment by sgibson91** [comment]
Hi @zaneselvans 👋🏻 Do you mind filling us in on how users of your hub are using dask/dask-gateway if at all so we can rectify this issue please?

**Comment by zaneselvans** [comment]
We primarily use dask to work with long time series that are stored in Parquet files. E.g. in [this example notebook](https://github.com/catalyst-cooperative/pudl-examples/blob/main/notebooks/03-pudl-parquet.ipynb). We've mainly been using it for convenient serialization of data processing. We haven't made use of dask clusters or dask-gateway yet.

Before the end of the year we should have finished re-factoring our larger data processing pipeline to use Prefect, at which point having access to a dask cluster for ongoing development could become helpful.

If I just add `dask_gateway` to our Docker image will that prevent this problem for now? I'll need to update it later this week anyway to provide access to a new version of our software and data.

**Comment by sgibson91** [comment]
> If I just add dask_gateway to our Docker image will that prevent this problem for now?

Thanks @zaneselvans - yes it will!

**Comment by sgibson91** [comment]
Yuvi opened a PR to address this here https://github.com/2i2c-org/infrastructure/issues/586#issuecomment-964509163

**Comment by zaneselvans** [comment]
Sorry, I've merged that PR in if you want to update the docker image our hub is pointing at once it builds. I will updating in again shortly.

**Comment by sgibson91** [comment]
Thanks @zaneselvans! This issue was fixed by merging PR #830 and also bumping the image tag in the hub's configurator
